strutil/quantity/example_test.go

```diff
@@ -115,3 +115,2866 @@ func ExampleFormatDuration() {
 	// " 18Ey"
 	// "ages!"
 }
+
+var dataset = []float64{
+	-math.MaxFloat64,
+	float64(math.MinInt64),
+	-time.Duration(3 * time.Nanosecond).Seconds(),
+	time.Duration(3 * time.Nanosecond).Seconds(),
+	time.Duration(36 * time.Microsecond).Seconds(),
+	time.Duration(36*time.Microsecond + 3*time.Nanosecond).Seconds(),
+	time.Duration(420*time.Millisecond + 36*time.Microsecond + 3*time.Nanosecond).Seconds(),
+	time.Duration(430 * time.Millisecond).Seconds(),
+	time.Duration(5155 * time.Millisecond).Seconds(),
+	time.Duration(time.Minute + 2*time.Second).Seconds(),
+	time.Duration(124 * time.Minute / 10).Seconds(),
+	time.Duration(2*time.Hour + 29*time.Minute).Seconds(),
+	time.Duration(10*time.Hour + 9*time.Minute).Seconds(),
+	time.Duration(10*time.Hour + 30*time.Minute).Seconds(),
+	time.Duration(10*time.Hour + 30*time.Minute + 2*time.Second).Seconds(),
+	time.Duration(11*time.Hour + 2*time.Minute).Seconds(),
+	time.Duration(22*time.Hour + 59*time.Minute + 59*time.Second).Seconds(),
+	time.Duration(23*time.Hour + 59*time.Minute + 59*time.Second).Seconds(),
+	time.Duration(30 * time.Hour).Seconds(),
+	time.Duration(345 * time.Hour).Seconds(),
+	time.Duration(357 * time.Hour).Seconds(),
+	time.Duration(4272 * time.Hour).Seconds(),
+	time.Duration(4273*time.Hour + 30*time.Minute + 2*time.Second).Seconds(),
+	time.Duration(51368 * time.Hour).Seconds(),
+	time.Duration(math.MaxInt64 / 10).Seconds(),
+	time.Duration(math.MaxInt64).Seconds(),
+	float64(math.MaxUint64) * 365 * 24 * 60 * 60,
+	math.MaxFloat64,
+}
+
+func Inv(dt float64) {
+	fmt.Printf("%q\n", quantity.FormatDurationGeneric(dt,
+		quantity.Year,
+		quantity.Sec,
+		quantity.ShowAll,
+		quantity.TimeLeft,
+		true))
+}
+
+func ExampleInv() {
+
+	for _, dt := range dataset {
+		Inv(dt)
+	}
+
+	// Output:
+	// "inv!"
+	// "inv!"
+	// "inv!"
+	// "inv!"
+	// "inv!"
+	// "inv!"
+	// "inv!"
+	// "inv!"
+	// "inv!"
+	// "inv!"
+	// "inv!"
+	// "inv!"
+	// "inv!"
+	// "inv!"
+	// "inv!"
+	// "inv!"
+	// "inv!"
+	// "inv!"
+	// "inv!"
+	// "inv!"
+	// "inv!"
+	// "inv!"
+	// "inv!"
+	// "inv!"
+	// "inv!"
+	// "inv!"
+	// "inv!"
+	// "inv!"
+}
+
+// Spaces with verbose packing
+
+func SpaceAllTimeLeftRange1(dt float64) {
+	fmt.Printf("%q\n", quantity.FormatDurationGeneric(dt,
+		quantity.NSec,
+		quantity.Year,
+		quantity.ShowAll,
+		quantity.TimeLeft,
+		quantity.SpaceOn))
+}
+
+func ExampleSpaceAllTimeLeftRange1() {
+
+	for _, dt := range dataset {
+		SpaceAllTimeLeftRange1(dt)
```

Comment:
I would find it more readable if the body of the SpaceAllTimeLeftRange1() function was inlined here.

Reply:
Discussed. Agreed that "Example" unit tests work as expected and cover all these cases.